### ISSUE TITLE
Improve log namespacing documentation

### ISSUE THREAD

**Issue by s-at-ik** [vectordotdev/vector#21360]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

I started using log namespacing and it is a great feature allowing for a more predictable behaviour without losing information. However the documentation is scarce. As far as I can tell, the main source of information is the [blog post from june 2023 regarding log namespacing introduction](https://vector.dev/blog/log-namespacing/). 

### Attempted Solutions

It is possible to figure out most things trial and error, but that's rather time consuming and inefficient. There is also some documentation to be found on the development side (e.g. [adding namespacing to a source](https://github.com/vectordotdev/vector/blob/master/docs/tutorials/lognamespacing.md) ), but again that's just a poor solution to the problem at hand.

### Proposal

I believe the blog post mentioned previously should be incorporated as part of the main documentation.

Some other points which could use clarification in my opinion:

- type change for `.`  (see #21342 )
- metadata generated by components (documentation only details "non namespaced" metadata)
- how to use semantic meaning and what is its relationship with log namespacing
- how sinks handle metadata / the `%` object

### Version

vector 0.41.1 (x86_64-unknown-linux-gnu 745babd 2024-09-11 14:55:36.802851761)

**Comment by jszwedko** [comment]
Thanks @s-at-ik . We had had updating documentation as part of the path towards making log namespacing the default behavior, but this has been on the back burner for a while (too long!). I appreciate you opening the issue for tracking and noting bits you think are important.

**Comment by vbmithr** [comment]
I've also trying to use log namespacing and found the documentation scarce. I'm sending events in vector from a custom source using the protobuf native interface, and found that I cannot apparently set metadata. The syntax `%source_name.field` does not work apparently in my case even though I populate the `metadata_full` field in the log protobuf

**Comment by jszwedko** [comment]
Do you mean you can't use `%source_name.field` to set metadata in a `remap` transform after ingesting the data? Or something else? As you noted, if sending proto encoded events the metadata should be under a separate field: `metadata_full`.

**Comment by vbmithr** [comment]
Yeah, the mistake I've made is that I thought the source name was automatically set by vector, but in practice it's up to the source to put its attributes in a map like `metadata_full { value = Map(source_name, attrs) }` and not { value = attrs } directly, otherwise those attrs are set in toplevel in vector.

**Comment by richlv** [comment]
`log_namespace` apparently only living in a blog made me think it has been removed - which is not the case, right?

If namespace still is not compatible with disk buffering, it would be useful to resurrect note from https://github.com/vectordotdev/vector/commit/9e973857738b5e3bce1fc04d0b5dd9090c4afe63#diff-eccf853a25d6a0d6a796efd63fa966f4f4e54776bb12c944655e43ef0eed377b in the docs. It seems to be lost from any public docs I could find.

**Comment by pront** [comment]
> `log_namespace` apparently only living in a blog made me think it has been removed - which is not the case, right?

Hey @richlv, did you see the new pages dedicated to global Vector config options? 

The `log_namespace` option is documented here: https://vector.dev/docs/reference/configuration/schema/#schema.log_namespace

We also released a dedicated guide: https://vector.dev/guides/level-up/log_namespace/


> If namespace still is not compatible with disk buffering, it would be useful to resurrect note from [9e97385#diff-eccf853a25d6a0d6a796efd63fa966f4f4e54776bb12c944655e43ef0eed377b](https://github.com/vectordotdev/vector/commit/9e973857738b5e3bce1fc04d0b5dd9090c4afe63#diff-eccf853a25d6a0d6a796efd63fa966f4f4e54776bb12c944655e43ef0eed377b) in the docs. It seems to be lost from any public docs I could find.

Hmm, this is indeed not documented. Thanks for calling this out. We will add a warning.

**Comment by richlv** [comment]
> did you see the new pages dedicated to global Vector config options?

Thanks, I had seen that, but it being rather terse, it seemed not that sufficient - not much detail, including the default.
I also interpreted the "/guide/" one as being more blog-like - not quite describing the feature in detail, and potentially not being updated as things change(?).

Sorry if this is a bit offtopic, but overall, is Vector.dev documentation AI or otherwise generated?
Asking as there seem to be several confusing inconsistencies, not sure what's the take on bugreports about docs.

**Comment by pront** [comment]
> Thanks, I had seen that, but it being rather terse, it seemed not that sufficient - not much detail, including the default. I also interpreted the "/guide/" one as being more
blog-like - not quite describing the feature in detail,

The reference page is intentionally concise. The guides are more open ended and can explain behavior in detail.

> and potentially not being updated as things change(?).

Do keep in mind that this feature still has some rough edges and is in beta.

This is one of the main reasons why we always recommend testing your Vector pipeline carefully before deploying to a prod environment.

> Sorry if this is a bit offtopic, but overall, is Vector.dev documentation AI or otherwise generated? Asking as there seem to be several confusing inconsistencies, not sure
what's the take on bugreports about docs.

Reference documentation is generated from RustDocs. 
The vast majority of the documentation predates current AI tooling by years.

--- 

The original request has been addressed for the most part:

- Reference page: https://vector.dev/docs/reference/configuration/schema/#schema.log_namespace
- Guide: https://vector.dev/guides/level-up/log_namespace/
- An open PR (#25560) adds a warning about the disk buffer incompatibility (and other improvements)

The remaining items from the proposal (especially individual sink behavior) are quite broad to track under a single issue. Please open more focused GitHub issues to address any remaining problems. Contributions are warmly welcomed.

Per above, I am closing this issue. Thanks for the original report @s-at-ik!
